File: MyStagePass.Model/Models/Event.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyStagePass.Model.Models
{
	public class Event
	{
		public enum TicketType
		{
			Regular = 1,
			Vip = 2,
			Premium = 3
		}
		public int EventID { get; set; }
		public string? EventName { get; set; }
		public string? Description { get; set; }
		public int TotalTickets { get; set; }
		public int TicketsSold { get; set; }
		public int RegularPrice { get; set; }
		public int VipPrice { get; set; }
		public int PremiumPrice { get; set; }
		public int PerformerID { get; set; }
		public virtual Performer Performer { get; set; } = null!;
		public DateTime EventDate { get; set; }
		public int LocationID { get; set; }
		public virtual Location Location { get; set; } = null!;
		public int StatusID { get; set; }
		public virtual Status Status { get; set; } = null!;
		public int TotalScore { get; set; } //ukupan zbir ocjena iz recenzija
		public int RatingCount { get; set; } //broj ocjena recenzija
		public float RatingAverage { get; set; } //prosjecne ocjena svih recenzija
		public virtual ICollection<CustomerFavoriteEvent> FavoritedByCustomers { get; set; } = new List<CustomerFavoriteEvent>();
		public virtual ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
		public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();

		[NotMapped]
		public int TicketsAvailable => TotalTickets - TicketsSold;
		public bool HasAvailableTickets(int requestedAmount)
		{
			return TicketsAvailable >= requestedAmount;
		}

		[NotMapped]
		public string TimeStatus
		{
			get
			{
				if (EventDate < DateTime.Now)
					return "Ended";
				else
					return "Upcoming";
			}
		}
	}
}

File: UI/mystagepass_mobile/lib/models/Event/event.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mystagepass_mobile/models/Status/status.dart';
import '../Performer/performer.dart';
import '../Location/location.dart';

part 'event.g.dart';

@JsonSerializable()
class Event {
  int? eventID;
  String? eventName;
  int? ticketsSold;
  int? totalTickets;
  DateTime? eventDate;
  String? timeStatus;
  int? regularPrice;
  int? vipPrice;
  int? premiumPrice;
  double? ratingAverage;
  int? ratingCount;
  int? userRating;
  String? description;

  Performer? performer;
  Location? location;
  Status? status;

  Event({
    this.eventID,
    this.eventName,
    this.ticketsSold,
    this.totalTickets,
    this.eventDate,
    this.timeStatus,
    this.regularPrice,
    this.vipPrice,
    this.premiumPrice,
    this.performer,
    this.location,
    this.ratingAverage,
    this.ratingCount,
    this.userRating,
    this.status,
    this.description,
  });

  factory Event.fromJson(Map<String, dynamic> json) => _$EventFromJson(json);
  Map<String, dynamic> toJson() => _$EventToJson(this);
}
